src/foam/core/stdlib.js

```diff
@@ -697,6 +698,9 @@ foam.typeOf = (function() {
   };
 })();
 
+foam.typeOrder = [ foam.Undefined, foam.Null, foam.Array,
```

Comment:
I think this should go from "simplest type" to "most complex type". To me, this would be:

`undefined, null, boolean, number, string, array, function, date, object, fobject`

I'm unsure about the ordering of `string`/`array` and `function`/`date`, but I think the others are "genuinely simpler/more complex".

If using this order introduces problems we should discuss the nature of the problems.